collect_app/src/test/java/org/odk/collect/android/formmanagement/download/ServerFormDownloaderTest.java

```diff
@@ -818,6 +818,59 @@ public void whenFormLacksEntityAttachmentsAndEntityBlock_downloadsAndSavesFormAs
         assertThat(form.usesEntities(), is(false));
     }
 
+    @Test
+    public void whenFormAlreadyDownloadedDoesNotHaveEntities_andHasNewMediaFilesWithEntities_downloadsAndMarkTheFormAsAFormThatUsesEntities() throws Exception {
+        String xform = createXFormBody("id", "version");
+        ServerFormDetails serverFormDetails = new ServerFormDetails(
+                "Form",
+                "http://downloadUrl",
+                "id",
+                "version",
+                Md5.getMd5Hash(new ByteArrayInputStream(xform.getBytes())),
+                true,
+                false,
+                new ManifestFile("", List.of(
+                        new MediaFile("file1", Md5.getMd5Hash(new ByteArrayInputStream("contents".getBytes())), "http://file1")
+                )));
+
+        FormSource formSource = mock(FormSource.class);
+        when(formSource.fetchForm("http://downloadUrl")).thenReturn(new ByteArrayInputStream(xform.getBytes()));
+        when(formSource.fetchMediaFile("http://file1")).thenReturn(new ByteArrayInputStream("contents".getBytes()));
+
+        ServerFormDownloader downloader = new ServerFormDownloader(formSource, formsRepository, cacheDir, formsDir.getAbsolutePath(), FormMetadataParser.INSTANCE, clock, entitiesRepository);
+
+        // Initial download
+        downloader.downloadForm(serverFormDetails, null, null);
+
+        List<Form> allForms = formsRepository.getAll();
+        assertThat(allForms.size(), is(1));
+        Form form = allForms.get(0);
+        assertThat(form.usesEntities(), is(false));
+
+        ServerFormDetails serverFormDetailsUpdatedMediaFile = new ServerFormDetails(
+                "Form",
+                "http://downloadUrl",
+                "id",
+                "version",
+                Md5.getMd5Hash(new ByteArrayInputStream(xform.getBytes())),
+                false,
+                false,
+                new ManifestFile("", List.of(
+                        new MediaFile("file1", Md5.getMd5Hash(new ByteArrayInputStream("contents-updated".getBytes())), "http://file1", true)
```

Comment:
The contents don't necessarily need to be updated! It looks like when processing an update, if `type=entityList` is seen in the manifest, the list is processed to a database even when the contents haven't changed.

Reply:
Whenever the branch at https://github.com/getodk/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormUseCases.kt#L108 is hit, the form should be marked as entity-related.

Reply:
I briefly thought that an alternative might be to move the check for an Entity List up so that it only happens if the media file contents have changed but actually we can't do that right now because the md5 hash will never match the manifest hash. We're relying on HTTP caching to not duplicate downloads. This will be improved as part of broader form download improvements.

Reply:
Ah, I just actually looked at the details of how and where `entityAttachmentsDetected` is set and I believe that will work. The test should just be updated to make it clear that a content change is not necessary.

Too bad to do two passes over the manifest but we can likely approve that in another pass.

Reply:
Got it, that makes sense. I now see that there's a special branch just for having detected any Entity List which matches https://github.com/getodk/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormUseCases.kt#L108

Reply:
> Too bad there are two passes over the manifest but we can likely improve that as part of download improvements as well.

Now there is one see: https://github.com/getodk/collect/pull/6494/commits/1d92ec3d160c4624dacfc9a7684a2dbcee3c51df